### PR TITLE
doc: manifest and updates for CMSE terminology fixes

### PR DESCRIPTION
**Pull request by greg-fer**
- Manifest update for nrfxlib.
- Minor heading edit in sdk-nrf docs.

### CODE REVIEW

**Review by frkv — APPROVED** [review]
LGTM (pending correct SHA)

**Review by mia-ko — APPROVED** [review]

File: west.yml
Comment (rlubos) [inline west.yml:118]: Please update SHA: `dc28a4f6addb944403c872d69de6f6dc428a8b31`

**Review by rlubos — APPROVED** [review]

**Comment by rlubos** [comment]
Doc update only, no need to wait for full CI.

**Pull request merged**
